classes/06class/exercises/c06-serverless01/alanlima/Makefile

```diff
@@ -0,0 +1,85 @@
+RUNNER ?= docker-compose run --rm -T
+
+init: pull-required-images
+	@$(RUNNER) terraform init
+.PHONY:init
+
+apply: 
+	@$(RUNNER) terraform apply
+.PHONY: apply
+
+deploy: init pack_lambda
+	@$(RUNNER) terraform apply -auto-approve
+.PHONY: deploy
+
+clean:
+	@$(RUNNER) terraform destroy -auto-approve
+	@rm -f ./terraform/lambda.zip
+.PHONY: clean
+
+pack_lambda:
+	@$(RUNNER) toolbox make _pack_lambda
+.PHONY:pack_lambda
+
+_pack_lambda:
+	@cd ./src && zip lambda.zip main.py
+	@mv ./src/lambda.zip ./terraform/lambda.zip
+.PHONY:_pack_lambda
+
+warm-up:
+	@echo "\e[32mCreating new customer...\e[0m \n"
+	@$(MAKE) create-customer FIRST_NAME="Linus" LAST_NAME="Torvalds" EMAIL="fake@mail.com"  
+	@echo "\n \e[32mListing table content\e[0m \n"
+	@$(MAKE) list-customers
+.PHONY: warm-up
+
+create-customer: pull-required-images
+	@$(eval API_KEY=$(shell $(RUNNER) jq -r ".outputs[\"api_key\"].value" ./terraform/terraform.tfstate))
+	@$(eval API_URL=$(shell $(RUNNER) jq -r ".outputs[\"api_url\"].value" ./terraform/terraform.tfstate))
+
+	@$(eval CURL_REQUEST="curl -v -POST \
+		-H \"X-API-Key: $(API_KEY)\" \
+		-H \"Content-type: application/json\" \
+		--url \"$(API_URL)/customers\" \
+		-d \'{ \"firstname\": \"$(FIRST_NAME)\", \"lastname\": \"$(LAST_NAME)\", \"email\": \"$(EMAIL)\" }\'")
+
+	@echo "Running cURL command: $(CURL_REQUEST)"
+
+	@$(shell echo "$(CURL_REQUEST)")
```

Comment:
Should it be in a bash script instead of Makefile?

Reply:
Done, moved a few scripts to bash script 😛